pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -415,37 +415,36 @@ func (f *framework) RunFilterPlugins(
 	state *CycleState,
 	pod *v1.Pod,
 	nodeInfo *schedulernodeinfo.NodeInfo,
-) (finalStatus *Status) {
+) PluginToStatus {
+	var firstFailedStatus *Status
 	if state.ShouldRecordFrameworkMetrics() {
 		startTime := time.Now()
 		defer func() {
-			f.metricsRecorder.observeExtensionPointDurationAsync(filter, finalStatus, metrics.SinceInSeconds(startTime))
+			f.metricsRecorder.observeExtensionPointDurationAsync(filter, firstFailedStatus, metrics.SinceInSeconds(startTime))
 		}()
 	}
+	statuses := make(PluginToStatus, 1)
 	for _, pl := range f.filterPlugins {
 		pluginStatus := f.runFilterPlugin(ctx, pl, state, pod, nodeInfo)
+		if len(statuses) == 0 {
```

Comment:
If you define `statuses := make(PluginToStatus, 1)`, doesn't that initiate with a length of 1 by default? Maybe L426 should be `make(PluginToStatus, 0, 1)`

Reply:
no, that is for slice, but I think by default go allocates a small number anyways, so omitted.